app/assets/javascripts/dragdrop.js

```diff
@@ -63,6 +65,13 @@ jQuery(function() {
           'nid':$D.nid
         },
         start: function(e) {
+          // 'start' function runs:
+          //   1. when user drag-and-drops image
+          //   2. when user clicks on upload button.
+          // for click-upload-button scenarios, it's important to set $D.selected here, because the 'drop' listener above doesn't run in those:
+          $D.selected = $(e.target).closest('div.comment-form-wrapper');
+          // the above line is redundant in drag & drop, because it's assigned in 'drop' listener too.
+          // on /wiki/new & /wiki/edit, $D.selected will = undefined from this assignment
```

Comment:
This is a major change: assign `$D.selected` in the `start` upload function.

As a reminder, this function runs on both `click` and `drop` image upload events. It's important to assign `$D.selected` here for `click` events, because the `drop` listener doesn't run for those.

I left a comment in the code about this, but I'm thinking it would be good down the road to not have to assign `$D.selected` twice in a row for `drop` events. Not sure how to do that yet.